src/app-layer-ftp.c

```diff
@@ -561,29 +562,34 @@ static AppLayerResult FTPParseRequest(Flow *f, void *ftp_state, AppLayerParserSt
 
 static int FTPParsePassiveResponse(FtpState *state, const uint8_t *input, uint32_t input_len)
 {
-    uint16_t dyn_port = rs_ftp_pasv_response(input, input_len);
-    if (dyn_port == 0) {
+    uint16_t dyn_port;
+    char *dyn_port_str = SCFTPPortFromPasvResponse(input, input_len, &dyn_port);
+    if (dyn_port_str == NULL) {
         return -1;
     }
     SCLogDebug("FTP passive mode (v4): dynamic port %"PRIu16"", dyn_port);
     state->active = false;
     state->dyn_port = dyn_port;
     state->curr_tx->dyn_port = dyn_port;
+    state->curr_tx->dyn_port_str = dyn_port_str;
```

Comment:
You should check if `state->curr_tx->dyn_port_str` is not NULL to avoid the leak, right ?